com.unity.robotics.urdf-importer/Tests/Runtime/SensorTests.cs

```diff
@@ -0,0 +1,168 @@
+using System.Collections;
+using System.IO;
+using System.Collections.Generic;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using RosSharp.Urdf;
+using System.Xml;
+using System.Xml.Linq;
+
+public class SensorTests
+{
+    // A Test behaves as an ordinary method
+    private TextReader GetSample()
+    {
+        string sampleDoc = "<sensor name='my_camera' type='camera'>\n" +
+                            "  <camera>\n" +
+                            "    <save enabled=\"true\">\n" +
+                            "      <path>/tmp/camera_save_tutorial</path>\n" +
+                            "    </save>\n" +
+                            "    <horizontal_fov>1.047</horizontal_fov>\n" +
+                            "    <image>\n" +
+                            "      <width>1920</width>\n" +
+                            "      <height>1080</height>\n" +
+                            "    </image>\n" +
+                            "    <clip>\n" +
+                            "      <near>0.1</near>\n" +
+                            "      <far>100</far>\n" +
+                            "    </clip>\n" +
+                            "  </camera>\n" +
+                            "  <always_on>1</always_on>\n" +
+                            "  <update_rate>30</update_rate>\n" +
+                            "  <plugin name='test_plugin' filename='test_filename'>\n" +
+                            "  </plugin>\n" +
+                            "</sensor>\n";
+        return new StringReader(sampleDoc);
+    }
+    [Test]
+    public void TotalDataTest()
+    {
+        
+        int totalData = 13;
+        XDocument xdoc = XDocument.Load(GetSample());
+        Sensor sensor = new Sensor(xdoc.Element("sensor"));
+        Assert.AreEqual(totalData,sensor.elements.Count);
+    }
+
+    [Test]
+    public void NumberofAttributeTest()
+    {
+        int totalAttributes = 5;
+        int numberofAttributes = 0;
+        XDocument xdoc = XDocument.Load(GetSample());
+        Sensor sensor = new Sensor(xdoc.Element("sensor"));
+        foreach(var key in sensor.elements.Keys)
+        {
+            if(key.Contains("@"))
+            {
+                numberofAttributes++;
+            }
+        }
+        Assert.AreEqual(totalAttributes , numberofAttributes);
+    }
+
+    [Test]
+    public void NumberofElementsTest()
+    {
+        int totalElements = 8;
+        int numberofElements = 0;
+        XDocument xdoc = XDocument.Load(GetSample());
+        Sensor sensor = new Sensor(xdoc.Element("sensor"));
+        foreach(var key in sensor.elements.Keys)
+        {
+            if(!(key.Contains("@")))
+            {
+                numberofElements++;
+            }
+        }
```

Comment:
Instead of parsing the xml, you should pass the xml to the Sensor class, and then get the elements from sensor.elements